File: VB/GridProject/ExtendedGridView/ExtendedGridView.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Controls
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.Utils

Namespace ExtendedGridViewSpace
	<System.ComponentModel.DesignerCategory("")> _
	Public Class ExtendedGridView
		Inherits GridView

		Public Sub New()
		End Sub
		Public Sub New(ByVal ownerGrid As GridControl)
			MyBase.New(ownerGrid)

		End Sub

		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "ExtendedGridView"
			End Get
		End Property

		Public Overrides Function GetShowEditorMode() As EditorShowMode
			Dim col As ExtendedGridColumn = TryCast(FocusedColumn, ExtendedGridColumn)
			If col IsNot Nothing Then
				Return col.EditorShowMode
			End If
			Return MyBase.GetShowEditorMode()
		End Function

		Protected Overrides Function CreateColumnCollection() As GridColumnCollection
			Dim columns As GridColumnCollection = New ExtendedGridColumnCollection(Me)
			Return columns
		End Function
	End Class
End Namespace

File: CS/GridProject/ExtendedGridView/ExtendedGridView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Controls;
using DevExpress.XtraGrid.Columns;
using DevExpress.Utils;

namespace ExtendedGridViewSpace
{
    [System.ComponentModel.DesignerCategory("")]
    public class ExtendedGridView : GridView
    {

        public ExtendedGridView()
        {}
        public ExtendedGridView(GridControl ownerGrid)
            : base(ownerGrid)
        {
            
        }

        protected override string ViewName
        {
            get
            {
                return "ExtendedGridView";
            }
        }

        public override EditorShowMode GetShowEditorMode(int rowHandle)
        {
            ExtendedGridColumn col = FocusedColumn as ExtendedGridColumn;
            if (col != null)
            {
                return col.EditorShowMode;
            }
            return base.GetShowEditorMode();
        }

        protected override GridColumnCollection CreateColumnCollection()
        {
            GridColumnCollection columns = new ExtendedGridColumnCollection(this);
            return columns;
        }
    }
}